Sources/MapboxCoreNavigation/PassiveLocationDataSource.swift

```diff
@@ -22,16 +22,24 @@ open class PassiveLocationDataSource: NSObject {
         self.directions = directions
         
         let settingsProfile = SettingsProfile(application: ProfileApplication.kMobile, platform: ProfilePlatform.KIOS)
-        self.navigator = try! Navigator(profile: settingsProfile,
-                                        config: NavigatorConfig(),
-                                        customConfig: "",
-                                        tilesConfig: TilesConfig())
+        let config = try! ConfigFactory.build(for: settingsProfile, config: NavigatorConfig(), customConfig: "")
+        let runLoopExecutor = try! RunLoopExecutorFactory.build()
+        let historyRecorder = try! HistoryRecorderHandle.build(forConfig: config)
+        let cache = try! CacheFactory.build(for: TilesConfig(), config: config, runLoop: runLoopExecutor, historyRecorder: historyRecorder)
+        self.graphAccessor = try! GraphAccessor(cache: cache)
+        self.navigator = try! Navigator(config: config, runLoopExecutor: runLoopExecutor, cache: cache, historyRecorder: historyRecorder)
```

Comment:
This will look quite different once #2823 lands. We’ll need to account for the need for a GraphAccessor that depends on the cache.

Reply:
I’ve rebased this branch atop release-v2.0, which now includes the changes in #2823.